haskell/toolchain.bzl

```
@@ -212,9 +213,9 @@ def _haskell_toolchain_impl(ctx):
 
     if ctx.attr.asterius_binaries:
         tools_config = asterius_tools_config(
-            exec_cc_toolchain = find_cc_toolchain(ctx),
+            exec_cc_toolchain = ctx.attr._exec_cc_toolchain[platform_common.ToolchainInfo],
```

```suggestion
            exec_cc_toolchain = ctx.attr._exec_cc_toolchain[cc_common.CcToolchainInfo],
```
Otherwise this fails with Bazel 5.

You can relatively easily test against Bazel 5 in bindist mode by using bazelisk with the env-var `USE_BAZEL_VERSION=5.0.0rc2`.